### PR TITLE
Layertree one wms

### PR DESCRIPTION
**Pull request by ger-benjamin**
Rel to: #576 and https://github.com/camptocamp/c2cgeoportal/issues/1970

Example: https://ger-benjamin.github.io/ngeo/master/examples/contribs/gmf/layertree.html
And: https://ger-benjamin.github.io/ngeo/master/examples/contribs/gmf/apps/mobile/index.html

Please review.
(Ignore coveralls ? I think it is not not happy because I've deleted some tests (but relatives methods too))

### CODE REVIEW

**Comment by fgravin** [comment]
i'll review it later in the evening

File: contribs/gmf/src/directives/layertree.js [hunk 1]
```diff
@@ -149,87 +149,272 @@ gmf.LayertreeController = function($http, $sce, ngeoCreatePopup,
...
-  layer = this.layerHelper_.findLayer(this.existingLayers_,
-      this.layerHelper_.makeHelperID(layerURL, layerName));
   if (goog.isDefAndNotNull(layer)) {
-    return layer;
+    layer.set('querySourceId', node.id);
+    layer.set('layerName', node.name);
+    // The layer must be upper than the background
+    layer.setZIndex(1);
+    // Add the new layer on the map but behind (before) others layers
+    this.map.getLayerGroup().getLayers().insertAt(0, layer);
```
Comment (fgravin) [inline contribs/gmf/src/directives/layertree.js]: `this.map.getLayers()` is more understandable.

**Comment by fgravin** [comment]
We should add some tests on gmf layertree directive methods.
The logic becomes a bit complex

File: contribs/gmf/src/directives/layertree.js (hunk 1 above)
Comment (ger-benjamin) [inline contribs/gmf/src/directives/layertree.js]: I'm note sur that that's completely the same things... (I'm not sur I've the `insertAt` method with your idea). But I'll try it.

**Comment by ger-benjamin** [comment]
> We should add some tests on gmf layertree directive methods.

Ok, I'll do that.

> The logic becomes a bit complex

I know. That's hard to understand. But there is a lot of possible case to handle and I've also must take the performance in count the performance and future possibilities like activate layers by a directlink or by the search.... I've don't found a better solution... (And, at the end, I like this one).

**Comment by fgravin** [comment]
Ok you can merge.
Thanks for the work

**Pull request merged**